src/get-started/first-java-service.md

```diff
@@ -0,0 +1,605 @@
+  For `String` keys, we use a standard serializer. For `Vehicle`s, which
+  are Protocol Buffers messages, we use a corresponding serializer for messages
+  of `Vehicle` type.
+
+!!! tip
+    As the [`StandardSerializers.protobuf`][protobuf-serializer] uses reflection
+    to look up the needed methods in the message class, it is recommended
+    to instantiate a protobuf serializer once for each type
+    and keep it in a `static` field, e.g.:
+
+    ```java
+    private static final Serializer<Vehicle> VEHICLE_SERIALIZER =
+        StandardSerializers.protobuf(Vehicle.class);
+    ```
+
+[protobuf-serializer]: https://exonum.com/doc/api/java-binding/0.10.0/com/exonum/binding/common/serialization/StandardSerializers.html#protobuf(java.lang.Class)
+
+??? fail "Cannot import `Vehicle` class?"
+    In some IDEs, e.g. IntelliJ IDEA, it might be needed to manually
+    mark `car-registry-messages/target/generated-sources/protobuf/java`
+    directory as "Generated Sources Root" to be able to import the generated
+    classes:
+
+    1. Open context menu (right click on the directory)
+    2. "Mark Directory As" > "Generated Sources Root"
+
+Compile the project:
+
+```shell
+mvn compile
+```
+
+### Service Transactions
+
+Our service needs two operations updating its state:
+
+- Add a new vehicle entry to the registry
+- Transfer the ownership over the vehicle to another user.
+
+Modifying operations in Exonum are called [_transactions_](../glossary.md#transaction).
+Transactions are implemented as methods in a service class — a class implementing
+[`Service`][service-interface] interface. A transaction method must be annotated
+with [`@Transaction`][transaction-annotation] annotation.
+
+[service-interface]: https://exonum.com/doc/api/java-binding/0.10.0/com/exonum/binding/core/service/Service.html
+[transaction-annotation]: https://exonum.com/doc/api/java-binding/0.10.0/com/exonum/binding/core/transaction/Transaction.html
+
+Our project already has a template service named `MyService`.
+
+#### Add Vehicle Transaction
+
+First, let's define the transaction arguments. As Exonum expects transaction
+arguments in a serialized form, we will define the arguments as a protobuf
+message.
+
+Add a new file `transactions.proto` in `car-registry-messages`
+(`car-registry-messages/src/main/proto/example/vehicle`).
+
+Then add the message definition of the _Add Vehicle_ transaction:
+
+<!--codeinclude-->
+[](../../code-examples/java/exonum-java-binding/tutorials/car-registry/car-registry-messages/src/main/proto/example/vehicle/transactions.proto)
+inside_block:ci-add-vehicle
+<!--/codeinclude-->
+
+Compile the message:
+
+```shell
+mvn generate-sources
+```
+
+---
+
+Next, let's write the transaction method. Navigate to `MyService`
+and add the following method:
+
+<!--codeinclude-->
+[](../../code-examples/java/exonum-java-binding/tutorials/car-registry/car-registry-service/src/main/java/com/example/car/MyService.java)
```

Comment:
Micro-nit: This code snippet introduces a horizontal scrollbar on the full page layout, seemingly because of the line starting with `var errorDescription = `.